docs/building-apps/ingest-sdk/ingestion-pipeline-code.mdx

```diff
@@ -0,0 +1,290 @@
+  }
+
+  closeTime := uint(ledgerCloseMeta.LedgerHeaderHistoryEntry().Header.ScpValue.CloseTime)
+
+  // scan all transactions in a ledger for payments to derive new model from
+  transaction, err := ledgerTxReader.Read()
+  for ; err == nil; transaction, err = ledgerTxReader.Read() {
+    for _, op := range transaction.Envelope.Operations() {
+      switch op.Body.Type {
+      case xdr.OperationTypePayment:
+        networkPayment := op.Body.MustPaymentOp()
+        myPayment := AppPayment{
+          Timestamp:       closeTime,
+          BuyerAccountId:  networkPayment.Destination.Address(),
+          SellerAccountId: op.SourceAccount.Address(),
+          AssetCode:       networkPayment.Asset.StringCanonical(),
+          Amount:          amount.String(networkPayment.Amount),
+        }
+        jsonBytes, err := json.Marshal(myPayment)
+        if err != nil {
+          return err
+        }
+
+        for _, processor := range transformer.processors {
+          processor.Process(ctx, Message{Payload: jsonBytes})
+        }
+      }
+    }
+  }
+  if err != io.EOF {
+    return errors.Wrapf(err, "failed to read transaction from ledger %v", ledgerCloseMeta.LedgerSequence())
+  }
+  return nil
+}
+
+type CaptiveCoreInboundAdapter struct {
+  TomlParams                ledgerbackend.CaptiveCoreTomlParams
+  processors                []Processor
+  historyArchiveURLs        []string
+  coreConfigNetworkTemplate []byte
+}
+
+func (adapter *CaptiveCoreInboundAdapter) Subscribe(receiver Processor) {
+  adapter.processors = append(adapter.processors, receiver)
+}
+
+func (adapter *CaptiveCoreInboundAdapter) Run(ctx context.Context) error {
+  // Setup captive core config to use the Pubnet network
+  captiveCoreToml, err := ledgerbackend.NewCaptiveCoreTomlFromData(adapter.coreConfigNetworkTemplate, adapter.TomlParams)
+  if err != nil {
+    return errors.Wrap(err, "error creating captive core toml")
+  }
+
+  captiveConfig := ledgerbackend.CaptiveCoreConfig{
+    BinaryPath:         adapter.TomlParams.CoreBinaryPath,
+    HistoryArchiveURLs: adapter.TomlParams.HistoryArchiveURLs,
+    Context:            ctx,
+    Toml:               captiveCoreToml,
+  }
+
+  // Create a new captive core backend, the origin of ingestion pipeline
+  captiveBackend, err := ledgerbackend.NewCaptive(captiveConfig)
+  if err != nil {
+    return errors.Wrap(err, "error creating captive core instance")
+  }
+
+  // Create a client to the network's history archives
+  historyAchive, err := historyarchive.NewArchivePool(adapter.historyArchiveURLs, historyarchive.ArchiveOptions{
+    ConnectOptions: storage.ConnectOptions{
+      UserAgent: "my_app",
+      Context:   ctx,
+    },
+  })
+
+  if err != nil {
+    return errors.Wrap(err, "error creating history archive client")
+  }
+
+  // Acquire the most recent ledger on network
+  latestNetworkLedger, err := datastore.GetLatestLedgerSequenceFromHistoryArchives(historyAchive)
```

Comment:
Completely unrelated to this PR, but I was wondering if the datastore is the right place for `GetLatestLedgerSequenceFromHistoryArchives`?

Reply:
I think it's been moved around a bit as a util function, and packages between what's common ingestion sdk and just ledger exporter have shuffled, this does seem like it could potentially be on the ha interface `historyatchive.ArchiveInterface.GetLatestLedgerSequence()`